pkg/didcomm/protocol/issuecredential/states.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package issuecredential
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+)
+
+const (
+	// common states
+	stateNameStart      = "start"
+	stateNameAbandoning = "abandoning"
+	stateNameDone       = "done"
+	stateNameNoop       = "noop"
+
+	// states for Issuer
+	stateNameProposalReceived = "proposal-received"
+	stateNameOfferSent        = "offer-sent"
+	stateNameRequestReceived  = "request-received"
+	stateNameCredentialIssued = "credential-issued"
+
+	// states for Holder
+	stateNameProposalSent       = "proposal-sent"
+	stateNameOfferReceived      = "offer-received"
+	stateNameRequestSent        = "request-sent"
+	stateNameCredentialReceived = "credential-received"
+)
+
+// state action for network call
+type stateAction func(messenger service.Messenger) error
+
+type metaData struct{}
+
+// nolint: deadcode,unused
+// represents zero action function
+func zeroAction(_ service.Messenger) error { return nil }
```

Comment:
Is this no-op function for state transition ? Also, can we remove this if this is not used in the PR.

Reply:
Removed!